### ISSUE TITLE
AppDrawer not working on `5.0.0-beta.2`

### ISSUE THREAD

**Issue by DynxstyGIT** [jb3rndt/PersistentBottomNavBarV2#76]
Hey, I migrated my code to the newest beta and noticed that my AppDrawer stopped working (its not opening). Is there any fix for that?

```dart
return Scaffold(
      drawer: _buildDrawer(),
      body: PersistentTabView(...)
);

Drawer _buildDrawer() {
    return Drawer(...);
}
```

I use `Scaffold.of(context).openDrawer()` to open the Drawer

**Comment by DynxstyGIT** [comment]
Alright, seems like I fixed it by adding a `GlobalKey<ScaffoldState>` to my Scaffold and then calling `GlobalKeys.scaffoldKey.currentState!.openDrawer()` to open the AppDrawer

Dunno if this is actually intended, so I'll leave this issue open for now.

**Comment by jb3rndt** [comment]
It doesnt work with your initial approach, because the PersistentTabView internally creates a Scaffold, everything is wrapped in. So the context you used chooses the closest Scaffold above it, which is the internal one. Your solution seems like a valid fix, but you could also use the global context for that.
Apart from that, I wouldnt recommend wrapping the PersistentTabView, but I guess you need that for the Drawer right? Do you want that drawer to be the same and accessible from every page? Or how exactly are you using that? Because the Drawer is a component, that wasnt properly handled in v4, but it isnt now either, because its difficult to integrate into everything.

**Comment by DynxstyGIT** [comment]
yeah, I need that extra Scaffold for my Drawer and yeah it should be accessible from every screen. 
I just wondered if this was intended, as it worked before.

**Comment by jb3rndt** [comment]
Okay, I have the following solution in mind:
As I already said, the `PersistentTabView` already creates a scaffold, so the `PersistentTabView` must not be wrapped in an additional scaffold. To provide a drawer, the `PersistentTabView` could accept a drawer like the standard Scaffold does. Then, the package would include a helper function (like `openMainDrawer()`) that opens this drawer.
Would that be a better solution for your Problem? Or do you have another idea?

**Comment by DynxstyGIT** [comment]
> Okay, I have the following solution in mind:
> 
> As I already said, the `PersistentTabView` already creates a scaffold, so the `PersistentTabView` must not be wrapped in an additional scaffold. To provide a drawer, the `PersistentTabView` could accept a drawer like the standard Scaffold does. Then, the package would include a helper function (like `openMainDrawer()`) that opens this drawer.
> 
> Would that be a better solution for your Problem? Or do you have another idea?

It would definitely remove the need to have a secondary scaffold, yas

**Comment by jb3rndt** [comment]
I just published version `5.0.0-beta.4` which allows you to pass a drawer to `PersistentTabView`.
